[Hub Generated] Review request for Microsoft.NotificationHubs to add version preview/2023-01-01-preview

kmiecikt opened the pull request:
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/Azure/azure-rest-api-specs/kmiecikt-notificationhubs-Microsoft.NotificationHubs-2023-01-01-preview...main/).
## ARM API Information (Control Plane)

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
      _The preview version has been deployed globally in May 2023._
  4. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
      _As soon as it is approved - preferably, June 2023._
  6. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

### Contribution checklist (MS Employees Only):
- [x] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version). Note that this doesn't apply if you are trying to merge a PR that was previously in the private repository.

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://eng.ms/docs/cloud-ai-platform/azure-core/azure-core-pm-and-design/trusted-platform-pm-karimb/service-lifecycle-and-actions-team/service-lifecycle-actions-team/apex/media/overview_breakingchanges).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](http://aka.ms/aprwiki))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

kmiecikt committed:
Adds base for updating Microsoft.NotificationHubs from version stable/2017-04-01 to version 2023-01-01-preview

kmiecikt committed:
Updates readme

kmiecikt committed:
Updates API version in new specs and examples

kmiecikt committed:
Notification Hubs: adding API version 2023-01-01-preview

kmiecikt committed:
Notification Hubs: udpating custom words

kmiecikt committed:
Notification Hubs: updating custom-words

kmiecikt commented:
Justification for failing checks:
- The Breaking Changes reported by the validation tool are mostly false alarms caused by very outdated previous swagger version. Detailed description is in scenario Azure Dev Ops scenario 22744028.
- The only new Swagger LintDiff error is not really new - the `subscriptionId` already appeared in both the path and body in the previous version.

kmiecikt commented:
> @kmiecikt Please fix CI checks errors (`LintDiff`), you can refer to this [comments](https://github.com/Azure/azure-rest-api-specs/pull/24072#issuecomment-1556739784) or https://aka.ms/ci-fix on how to fix it.

Hi. Regarding the LintDiff error: we cannot fix them without introducing a breaking change. The error recognized as new is this one:

> The 'subscriptionId' already appears in the path, please don't repeat it in the request body.
> Location: [Microsoft.NotificationHubs/preview/2023-01-01-preview/notificationhubs.json#L886](https://github.com/Azure/azure-rest-api-specs/blob/119a5cbe947d30b0ac7890f4c942d8c5a9515ef6/specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/preview/2023-01-01-preview/notificationhubs.json#L886)_

The problem already exist in the current public API version - the `subscriptionId` from the query is the same as `subscriptionId` from Notification Hub Namespace properties. This was a mistake made years ago, but we cannot remove the property now. We are going to start the deprecation process in the next version.

I think it started appearing as new error because we started using shared `subscriptionId` parameter in the new Swagger, while it was a local, per-operation parameter in the previous Swagger version.

v-xuto commented:
@kmiecikt Please contact `ARM Team` to review this `LintDiff` error. If they feel okay, please let them to add label `Approved-LintDiff`. If you don't know how to contact `ARM team`, you can check the third of the `ARM API Review Checklist` in the [first comment](https://github.com/Azure/azure-rest-api-specs/pull/24072#issue-1716936690).

v-xuto commented:
@kmiecikt You should contact `ARM Team` to review this `LintDiff` error. 
If they feel okay, please let them to add label `Approved-LintDiff`. If you don't know how to contact `ARM team`, you can check the third of the `ARM API Review Checklist` in the https://github.com/Azure/azure-rest-api-specs/pull/24072#issue-1716936690.

kmiecikt committed:
Changing  zoneRedundant (boolean) to zoneRedundancy (enum), making Network ACLs optional

kmiecikt committed:
Notification Hubs: updating examples

kmiecikt committed:
Notification Hubs: fixing property name

kmiecikt committed:
Notification Hubs: fixing examples

mentat9 commented:
> @kmiecikt You should contact `ARM Team` to review this `LintDiff` error. If they feel okay, please let them to add label `Approved-LintDiff`. If you don't know how to contact `ARM team`, you can check the third of the `ARM API Review Checklist` in the [#24072 (comment)](https://github.com/Azure/azure-rest-api-specs/pull/24072#issue-1716936690).

I'm adding the Approved-LintDiff label to approve the use of subscriptionId in the response body (since removing it is a breaking change). Note: I haven't reviewed the other linter warnings.

@v-xuto - I'm also not clear why this PR wasn't flagged for ARM review, maybe because BreakingChangeReviewRequired isn't cleared yet?

v-xuto commented:
/azp run

v-xuto commented:
@JeffreyRichter Please help review the breaking change in this PR.

v-xuto commented:
@kmiecikt  Please contact `ARM team` to review this PR. If they feel ok, they need to add label `ARMSignedOff`.

mentat9 commented:
>         "isAvailiable": {

ARM recommends enum over boolean for a bunch of reasons. Consider something like currentAvailability: { Available, Unavailable }

---
Refers to: specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/preview/2023-01-01-preview/notificationhubs.json:2140 in 4d9c53a. [](commit_id = 4d9c53a8c6e91980173a7a37008d0743fe08e6d4, deletion_comment = False)

mentat9 commented:
>         "name": {

Top-level property names are not allowed to be reused within the property bag. This property needs to be renamed.

---
Refers to: specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/preview/2023-01-01-preview/notificationhubs.json:2746 in 4d9c53a. [](commit_id = 4d9c53a8c6e91980173a7a37008d0743fe08e6d4, deletion_comment = False)

mentat9 commented:
>         "authorizationRules": {

Secrets must be nulled or blanked/emptied in GET requests. If these need to be retrieved, it needs to be done via a separate POST request (to allow for separate RBAC). Note, this applies to every property containing secrets/marked with x-ms-secret.

---
Refers to: specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/preview/2023-01-01-preview/notificationhubs.json:2758 in 4d9c53a. [](commit_id = 4d9c53a8c6e91980173a7a37008d0743fe08e6d4, deletion_comment = False)

mentat9 commented:
>     "Operation": {

Not needed if you get OperationListResult from common types.

---
Refers to: specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/preview/2023-01-01-preview/notificationhubs.json:2821 in 4d9c53a. [](commit_id = 4d9c53a8c6e91980173a7a37008d0743fe08e6d4, deletion_comment = False)

mentat9 commented:
>         "properties": {

Nonstandard property for Operation API. What is this for?

---
Refers to: specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/preview/2023-01-01-preview/notificationhubs.json:2833 in 4d9c53a. [](commit_id = 4d9c53a8c6e91980173a7a37008d0743fe08e6d4, deletion_comment = False)

mentat9 commented:
>     "OperationDisplay": {

Not needed if you use Operation definition from common-types

---
Refers to: specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/preview/2023-01-01-preview/notificationhubs.json:2843 in 4d9c53a. [](commit_id = 4d9c53a8c6e91980173a7a37008d0743fe08e6d4, deletion_comment = False)

mentat9 commented:
>     "OperationListResult": {

Use OperationListResult from v5 common types.

---
Refers to: specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/preview/2023-01-01-preview/notificationhubs.json:2869 in 4d9c53a. [](commit_id = 4d9c53a8c6e91980173a7a37008d0743fe08e6d4, deletion_comment = False)

mentat9 commented:
>         "InProgress",

How is InProgress different from Pending? Consider using Creating and Updating.

---
Refers to: specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/preview/2023-01-01-preview/notificationhubs.json:2904 in 4d9c53a. [](commit_id = 4d9c53a8c6e91980173a7a37008d0743fe08e6d4, deletion_comment = False)

mentat9 commented:
>           "type": "string",

Should be string enum.

---
Refers to: specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/preview/2023-01-01-preview/notificationhubs.json:3007 in 4d9c53a. [](commit_id = 4d9c53a8c6e91980173a7a37008d0743fe08e6d4, deletion_comment = False)

mentat9 commented:
>         "id": {

Id shouldn't be PATCHable.

---
Refers to: specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/preview/2023-01-01-preview/notificationhubs.json:2731 in 4d9c53a. [](commit_id = 4d9c53a8c6e91980173a7a37008d0743fe08e6d4, deletion_comment = False)

mentat9 commented:
>           "201": {

POST response status code with schema is always 200.

---
Refers to: specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/preview/2023-01-01-preview/notificationhubs.json:377 in 4d9c53a. [](commit_id = 4d9c53a8c6e91980173a7a37008d0743fe08e6d4, deletion_comment = False)

mentat9 commented:
>     "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.NotificationHubs/namespaces/{namespaceName}/notificationHubs/{notificationHubName}/pnsCredentials": {

This should be a verb or verb phrase, e.g. getPnsCredentials

---
Refers to: specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/preview/2023-01-01-preview/notificationhubs.json:730 in 4d9c53a. [](commit_id = 4d9c53a8c6e91980173a7a37008d0743fe08e6d4, deletion_comment = False)

mentat9 commented:
>               "$ref": "#/definitions/PnsCredentialsResource"

Sounds like this should return a collection/array...? If it just returns a single object, consider changing the description to clarify.

---
Refers to: specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/preview/2023-01-01-preview/notificationhubs.json:761 in 4d9c53a. [](commit_id = 4d9c53a8c6e91980173a7a37008d0743fe08e6d4, deletion_comment = False)

mentat9 commented:
>         "id": {

Id isn't PATCHable.

---
Refers to: specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/preview/2023-01-01-preview/notificationhubs.json:2530 in 4d9c53a. [](commit_id = 4d9c53a8c6e91980173a7a37008d0743fe08e6d4, deletion_comment = False)

mentat9 commented:
>         "enabled": {

Consider enum, e.g. activation: { Active, Disabled }

---
Refers to: specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/preview/2023-01-01-preview/notificationhubs.json:2556 in 4d9c53a. [](commit_id = 4d9c53a8c6e91980173a7a37008d0743fe08e6d4, deletion_comment = False)

mentat9 commented:
>             "name": "$count",

Use $top for this. Ref: [https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#pagination](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#pagination).

---
Refers to: specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/preview/2023-01-01-preview/notificationhubs.json:1055 in 4d9c53a. [](commit_id = 4d9c53a8c6e91980173a7a37008d0743fe08e6d4, deletion_comment = False)

mentat9 commented:
>             "name": "$count",

Use $top for this. Ref: [https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#pagination](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#pagination).

---
Refers to: specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/preview/2023-01-01-preview/notificationhubs.json:314 in 4d9c53a. [](commit_id = 4d9c53a8c6e91980173a7a37008d0743fe08e6d4, deletion_comment = False)

mentat9 commented:
>             "name": "$count",

Use $top for this. Ref: [https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#pagination](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#pagination).

---
Refers to: specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/preview/2023-01-01-preview/notificationhubs.json:1115 in 4d9c53a. [](commit_id = 4d9c53a8c6e91980173a7a37008d0743fe08e6d4, deletion_comment = False)

mentat9 commented:
>     "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.NotificationHubs/namespaces/{namespaceName}/pnsCredentials": {

Same feedback as above regarding action name and response format/description.

---
Refers to: specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/preview/2023-01-01-preview/notificationhubs.json:1465 in 4d9c53a. [](commit_id = 4d9c53a8c6e91980173a7a37008d0743fe08e6d4, deletion_comment = False)

mentat9 commented:
>         "description": "This is an asynchronous API, which sets the Azure-AsyncOperation header. \r\nCaller can use URI from that header to check status of the operation.",

Async requires x-ms-long-running-operation

---
Refers to: specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/preview/2023-01-01-preview/notificationhubs.json:1556 in 4d9c53a. [](commit_id = 4d9c53a8c6e91980173a7a37008d0743fe08e6d4, deletion_comment = False)

mentat9 commented:
>         "responses": {

Also need 200 and 204 for final response status codes.

---
Refers to: specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/preview/2023-01-01-preview/notificationhubs.json:1643 in 4d9c53a. [](commit_id = 4d9c53a8c6e91980173a7a37008d0743fe08e6d4, deletion_comment = False)

mentat9 commented:
>         "id": {

Don't reuse top-level property names within the property bag. Recommend resourceId  or notificationResourceId if it's a resourceId, or correct wording otherwise.

---
Refers to: specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/preview/2023-01-01-preview/notificationhubs.json:2361 in 4d9c53a. [](commit_id = 4d9c53a8c6e91980173a7a37008d0743fe08e6d4, deletion_comment = False)

mentat9 commented:
>       "description": "ARM linked notification properties.",

Use Azure instead of ARM in published strings/names.

---
Refers to: specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/preview/2023-01-01-preview/notificationhubs.json:2358 in 4d9c53a. [](commit_id = 4d9c53a8c6e91980173a7a37008d0743fe08e6d4, deletion_comment = False)

mentat9 commented:
>         "name": {

Don't reuse top-level property names within the property bag. Need to rename this property, e.g. metricName.

---
Refers to: specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/preview/2023-01-01-preview/notificationhubs.json:2391 in 4d9c53a. [](commit_id = 4d9c53a8c6e91980173a7a37008d0743fe08e6d4, deletion_comment = False)

mentat9 commented:
>         "aggregationType": {

Use string enum if this property takes a fixed set of values.

---
Refers to: specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/preview/2023-01-01-preview/notificationhubs.json:2411 in 4d9c53a. [](commit_id = 4d9c53a8c6e91980173a7a37008d0743fe08e6d4, deletion_comment = False)

mentat9 commented:
>         "fillGapWithZero": {

Recommend string enum instead of boolean, e.g. gapFill: { Empty, Zero }, or similar.

---
Refers to: specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/preview/2023-01-01-preview/notificationhubs.json:2430 in 4d9c53a. [](commit_id = 4d9c53a8c6e91980173a7a37008d0743fe08e6d4, deletion_comment = False)

mentat9 commented:
>           "description": "List of resource URIs to move.",

resource IDs?

---
Refers to: specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/preview/2023-01-01-preview/notificationhubs.json:2451 in 4d9c53a. [](commit_id = 4d9c53a8c6e91980173a7a37008d0743fe08e6d4, deletion_comment = False)

mentat9 commented:
>         "critical": {

Enum?

---
Refers to: specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/preview/2023-01-01-preview/notificationhubs.json:2561 in 4d9c53a. [](commit_id = 4d9c53a8c6e91980173a7a37008d0743fe08e6d4, deletion_comment = False)

mentat9 commented:
>         "subscriptionId": {

This should be removed it if's duplicate of the Uri parameter.

---
Refers to: specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/preview/2023-01-01-preview/notificationhubs.json:2566 in 4d9c53a. [](commit_id = 4d9c53a8c6e91980173a7a37008d0743fe08e6d4, deletion_comment = False)

mentat9 commented:
>         "createdAt": {

Use the one from top-level systemData.

---
Refers to: specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/preview/2023-01-01-preview/notificationhubs.json:2581 in 4d9c53a. [](commit_id = 4d9c53a8c6e91980173a7a37008d0743fe08e6d4, deletion_comment = False)

mentat9 commented:
>         "updatedAt": {

Use the one from top-level systemData.

---
Refers to: specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/preview/2023-01-01-preview/notificationhubs.json:2587 in 4d9c53a. [](commit_id = 4d9c53a8c6e91980173a7a37008d0743fe08e6d4, deletion_comment = False)

mentat9 commented:
>         "name": {

The name "name" is not allowed inside the property bag, since it's reserved for the top-level; however, IIRC this contract is spec'd across multiple RPs and can't be changed.

---
Refers to: specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/preview/2023-01-01-preview/notificationhubs.json:3137 in 4d9c53a. [](commit_id = 4d9c53a8c6e91980173a7a37008d0743fe08e6d4, deletion_comment = False)

mentat9 commented:
>         "outcome": {

Enum?

---
Refers to: specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/preview/2023-01-01-preview/notificationhubs.json:3202 in 4d9c53a. [](commit_id = 4d9c53a8c6e91980173a7a37008d0743fe08e6d4, deletion_comment = False)

mentat9 commented:
>         "id": {

Same comment/question. If this is part of NRP spec it can't be changed, otherwise needs to be renamed since id is reserved for top-level property.

---
Refers to: specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/preview/2023-01-01-preview/notificationhubs.json:3213 in 4d9c53a. [](commit_id = 4d9c53a8c6e91980173a7a37008d0743fe08e6d4, deletion_comment = False)

mentat9 commented:
>         "claimType": {

Enum?

---
Refers to: specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/preview/2023-01-01-preview/notificationhubs.json:3381 in 4d9c53a. [](commit_id = 4d9c53a8c6e91980173a7a37008d0743fe08e6d4, deletion_comment = False)

v-xuto commented:
@kmiecikt Please resolve the conflicting files first, then address Mentat9's comment.

kmiecikt committed:
Merge branch 'main' of https://github.com/Azure/azure-rest-api-specs into kmiecikt-notificationhubs-Microsoft.NotificationHubs-2023-01-01-preview

kmiecikt committed:
Notification Hubs: resolving some of PR comments

kmiecikt commented:
> > ```
> > "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.NotificationHubs/namespaces/{namespaceName}/notificationHubs/{notificationHubName}/pnsCredentials": {
> > ```
> 
> This should be a verb or verb phrase, e.g. getPnsCredentials
> 
> Refers to: specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/preview/2023-01-01-preview/notificationhubs.json:730 in [4d9c53a](https://github.com/Azure/azure-rest-api-specs/commit/4d9c53a8c6e91980173a7a37008d0743fe08e6d4). [](commit_id = [4d9c53a](https://github.com/Azure/azure-rest-api-specs/commit/4d9c53a8c6e91980173a7a37008d0743fe08e6d4), deletion_comment = False)

Unfortunately, we cannot change it - this operation already exists in the previous API versions.

In this version, we add similar API for Namespaces (the existing one is for Notification Hubs). To be consistent, we used the same name (`pnsCredentials`), even though `getPnsCredentials` would be better for new operations.

kmiecikt commented:
> > ```
> >           "$ref": "#/definitions/PnsCredentialsResource"
> > ```
> 
> Sounds like this should return a collection/array...? If it just returns a single object, consider changing the description to clarify.
> 
> Refers to: specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/preview/2023-01-01-preview/notificationhubs.json:761 in [4d9c53a](https://github.com/Azure/azure-rest-api-specs/commit/4d9c53a8c6e91980173a7a37008d0743fe08e6d4). [](commit_id = [4d9c53a](https://github.com/Azure/azure-rest-api-specs/commit/4d9c53a8c6e91980173a7a37008d0743fe08e6d4), deletion_comment = False)

This schema already exists in previous API versions. It stores multiple PNS credentials, but each is represented by a separate property.

kmiecikt commented:
> > ```
> >     "id": {
> > ```
> 
> Id isn't PATCHable.
> 
> Refers to: specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/preview/2023-01-01-preview/notificationhubs.json:2530 in [4d9c53a](https://github.com/Azure/azure-rest-api-specs/commit/4d9c53a8c6e91980173a7a37008d0743fe08e6d4). [](commit_id = [4d9c53a](https://github.com/Azure/azure-rest-api-specs/commit/4d9c53a8c6e91980173a7a37008d0743fe08e6d4), deletion_comment = False)

Fixed (for all PATCH operations)

kmiecikt commented:
> > ```
> >     "enabled": {
> > ```
> 
> Consider enum, e.g. activation: { Active, Disabled }
> 
> Refers to: specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/preview/2023-01-01-preview/notificationhubs.json:2556 in [4d9c53a](https://github.com/Azure/azure-rest-api-specs/commit/4d9c53a8c6e91980173a7a37008d0743fe08e6d4). [](commit_id = [4d9c53a](https://github.com/Azure/azure-rest-api-specs/commit/4d9c53a8c6e91980173a7a37008d0743fe08e6d4), deletion_comment = False)

This property exists in the previous API versions and we cannot change it now.
This property also duplicates the state from ProvisioningState, and we will likely start its deprecation in the next API version.

kmiecikt commented:
> > ```
> >         "name": "$count",
> > ```
> 
> Use $top for this. Ref: https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#pagination.
> 
> Refers to: specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/preview/2023-01-01-preview/notificationhubs.json:1055 in [4d9c53a](https://github.com/Azure/azure-rest-api-specs/commit/4d9c53a8c6e91980173a7a37008d0743fe08e6d4). [](commit_id = [4d9c53a](https://github.com/Azure/azure-rest-api-specs/commit/4d9c53a8c6e91980173a7a37008d0743fe08e6d4), deletion_comment = False)

This applies to all similar comments: the `$count` was (unfortunately) defined in the previous API version and customers are already using it.

Because of the huge gap between the previous and this API version (over 6 years), we now want to focus on:
- adding the new features (e.g. Private Link)
- fixing most of the Swagger issues (secrets annotations, read-only & immutable properties, correct schema for PATCH / PUT, etc)
- fixing previous Swagger definitions that were incorrect (i.e. did not match the actual responses)

Shortly after this version, we are planning to create another one which will start the deprecation process - this change is already too complex. For the `$count`, we will:
- Add the new `$top` property with the same meaning as `$count`.
- Mark the `$count` as deprecated (and remove after the required 3 years)
- Verify that only one of these properties is set in requests

kmiecikt committed:
Notification Hubs: adding x-ms-long-running-operation to one of Private Link operations

kmiecikt commented:
> > ```
> >     "description": "This is an asynchronous API, which sets the Azure-AsyncOperation header. \r\nCaller can use URI from that header to check status of the operation.",
> > ```
> 
> Async requires x-ms-long-running-operation
> 
> Refers to: specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/preview/2023-01-01-preview/notificationhubs.json:1556 in [4d9c53a](https://github.com/Azure/azure-rest-api-specs/commit/4d9c53a8c6e91980173a7a37008d0743fe08e6d4). [](commit_id = [4d9c53a](https://github.com/Azure/azure-rest-api-specs/commit/4d9c53a8c6e91980173a7a37008d0743fe08e6d4), deletion_comment = False)

Fixed

kmiecikt commented:
> > ```
> >     "id": {
> > ```
> 
> Don't reuse top-level property names within the property bag. Recommend resourceId or notificationResourceId if it's a resourceId, or correct wording otherwise.
> 
> Refers to: specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/preview/2023-01-01-preview/notificationhubs.json:2361 in [4d9c53a](https://github.com/Azure/azure-rest-api-specs/commit/4d9c53a8c6e91980173a7a37008d0743fe08e6d4). [](commit_id = [4d9c53a](https://github.com/Azure/azure-rest-api-specs/commit/4d9c53a8c6e91980173a7a37008d0743fe08e6d4), deletion_comment = False)

Fixed. I removed all schemas for ARM Linked Notifications - these are internal APIs that should not be exposed to customers.

kmiecikt commented:
> > ```
> >   "description": "ARM linked notification properties.",
> > ```
> 
> Use Azure instead of ARM in published strings/names.
> 
> Refers to: specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/preview/2023-01-01-preview/notificationhubs.json:2358 in [4d9c53a](https://github.com/Azure/azure-rest-api-specs/commit/4d9c53a8c6e91980173a7a37008d0743fe08e6d4). [](commit_id = [4d9c53a](https://github.com/Azure/azure-rest-api-specs/commit/4d9c53a8c6e91980173a7a37008d0743fe08e6d4), deletion_comment = False)

Fixed (by removing this schema)

kmiecikt commented:
> > ```
> >     "name": {
> > ```
> 
> Don't reuse top-level property names within the property bag. Need to rename this property, e.g. metricName.
> 
> Refers to: specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/preview/2023-01-01-preview/notificationhubs.json:2391 in [4d9c53a](https://github.com/Azure/azure-rest-api-specs/commit/4d9c53a8c6e91980173a7a37008d0743fe08e6d4). [](commit_id = [4d9c53a](https://github.com/Azure/azure-rest-api-specs/commit/4d9c53a8c6e91980173a7a37008d0743fe08e6d4), deletion_comment = False)

This also applies for all the properties of `ServiceSpecification`, `LogSpecification`, `MetricSpecification` and schemas used by these: the schema is defined by Azure Monitor / Shoebox Logs teams and we don't control the names / properties. They looked differently in the previous Swagger, because it was incorrect - the old schema did not match what has been returned by the API.

kmiecikt committed:
Notification Hubs: removing internal MoveResourceRequest

kmiecikt commented:
> > ```
> >       "description": "List of resource URIs to move.",
> > ```
> 
> resource IDs?
> 
> Refers to: specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/preview/2023-01-01-preview/notificationhubs.json:2451 in [4d9c53a](https://github.com/Azure/azure-rest-api-specs/commit/4d9c53a8c6e91980173a7a37008d0743fe08e6d4). [](commit_id = [4d9c53a](https://github.com/Azure/azure-rest-api-specs/commit/4d9c53a8c6e91980173a7a37008d0743fe08e6d4), deletion_comment = False)

Fixed by removing this schema - it's was not used (the Move API is internal)

kmiecikt commented:
> > ```
> >     "critical": {
> > ```
> 
> Enum?
> 
> Refers to: specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/preview/2023-01-01-preview/notificationhubs.json:2561 in [4d9c53a](https://github.com/Azure/azure-rest-api-specs/commit/4d9c53a8c6e91980173a7a37008d0743fe08e6d4). [](commit_id = [4d9c53a](https://github.com/Azure/azure-rest-api-specs/commit/4d9c53a8c6e91980173a7a37008d0743fe08e6d4), deletion_comment = False)

Applies also to the following properties mentioned in other comments (`createdAt`, `updatedAt`, `subscriptionId`): these properties already exist in the previous API versions. As mentioned in other comment, we are planning to start the deprecation process in the next API version.
-

kmiecikt commented:
> > ```
> >     "outcome": {
> > ```
> 
> Enum?
> 
> Refers to: specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/preview/2023-01-01-preview/notificationhubs.json:3202 in [4d9c53a](https://github.com/Azure/azure-rest-api-specs/commit/4d9c53a8c6e91980173a7a37008d0743fe08e6d4). [](commit_id = [4d9c53a](https://github.com/Azure/azure-rest-api-specs/commit/4d9c53a8c6e91980173a7a37008d0743fe08e6d4), deletion_comment = False)

This requires more explanation. The `DebugSend` is an existing API that allows to use Resource Provider API to use one of our Data Plane operations (Send Notification) in a synchronous way. It's mostly used by developers who want to verify Notification Hub settings by using Azure Portal to send notification to their own devices.

The response schema for that operation is more less copy of the Data Plane schema, which also defines the outcome as string. We may convert that to enum in the future, but we will have to update both Management and Data Plane APIs at similar time. We cannot do it right now.

kmiecikt commented:
> > ```
> >     "id": {
> > ```
> 
> Same comment/question. If this is part of NRP spec it can't be changed, otherwise needs to be renamed since id is reserved for top-level property.
> 
> Refers to: specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/preview/2023-01-01-preview/notificationhubs.json:3213 in [4d9c53a](https://github.com/Azure/azure-rest-api-specs/commit/4d9c53a8c6e91980173a7a37008d0743fe08e6d4). [](commit_id = [4d9c53a](https://github.com/Azure/azure-rest-api-specs/commit/4d9c53a8c6e91980173a7a37008d0743fe08e6d4), deletion_comment = False)

For all Private Link properties: we cannot change them - the schema is defined by the Azure Networking team and we only followed the spec.

kmiecikt commented:
> > ```
> >     "claimType": {
> > ```
> 
> Enum?
> 
> Refers to: specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/preview/2023-01-01-preview/notificationhubs.json:3381 in [4d9c53a](https://github.com/Azure/azure-rest-api-specs/commit/4d9c53a8c6e91980173a7a37008d0743fe08e6d4). [](commit_id = [4d9c53a](https://github.com/Azure/azure-rest-api-specs/commit/4d9c53a8c6e91980173a7a37008d0743fe08e6d4), deletion_comment = False)

We are planning to deprecate this property in next API version - the `claimType` is always null.

kmiecikt committed:
Merge branch 'main' of https://github.com/Azure/azure-rest-api-specs into kmiecikt-notificationhubs-Microsoft.NotificationHubs-2023-01-01-preview

kmiecikt committed:
Notification Hubs: fixing merge marker

kmiecikt committed:
Notification Hubs: fixing zoneRedundancy

kmiecikt committed:
Notification Hubs: updating Swagger

kmiecikt committed:
Notification Hubs: updating samples

ms-zhenhua commented:
> > > ```
> > >         "name": "$count",
> > > ```
> > 
> > 
> > Use $top for this. Ref: https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#pagination.
> > Refers to: specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/preview/2023-01-01-preview/notificationhubs.json:1055 in [4d9c53a](https://github.com/Azure/azure-rest-api-specs/commit/4d9c53a8c6e91980173a7a37008d0743fe08e6d4). [](commit_id = [4d9c53a](https://github.com/Azure/azure-rest-api-specs/commit/4d9c53a8c6e91980173a7a37008d0743fe08e6d4), deletion_comment = False)
> 
> This applies to all similar comments: the `$count` was (unfortunately) defined in the previous API version and customers are already using it.
> 
> Because of the huge gap between the previous and this API version (over 6 years), we now want to focus on:
> 
> * adding the new features (e.g. Private Link)
> * fixing most of the Swagger issues (secrets annotations, read-only & immutable properties, correct schema for PATCH / PUT, etc)
> * fixing previous Swagger definitions that were incorrect (i.e. did not match the actual responses)
> 
> Shortly after this version, we are planning to create another one which will start the deprecation process - this change is already too complex. For the `$count`, we will:
> 
> * Add the new `$top` property with the same meaning as `$count`.
> * Mark the `$count` as deprecated (and remove after the required 3 years)
> * Verify that only one of these properties is set in requests

@kmiecikt I did not find `$count` used in the previous version: https://github.com/Azure/azure-rest-api-specs/blob/main/specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/stable/2017-04-01/notificationhubs.json. Could you provide the link?

ms-zhenhua commented:
>     "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.NotificationHubs/namespaces/{namespaceName}/notificationHubs/{notificationHubName}/pnsCredentials": {

resolved it as already exists in 2017-04-01

---
In reply to: [1574766953](https://github.com/Azure/azure-rest-api-specs/pull/24072#issuecomment-1574766953) [](http://example.com/codeflow?ancestors=1574766953,1574766953)

---
Refers to: specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/preview/2023-01-01-preview/notificationhubs.json:730 in 4d9c53a. [](commit_id = 4d9c53a8c6e91980173a7a37008d0743fe08e6d4, deletion_comment = False)

ms-zhenhua commented:
>         "summary": "Approves or rejects Private Endpoint Connection.\r\nThis is a public API that can be called directly by Notification Hubs users.",

not create?

---
Refers to: specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/preview/2023-01-01-preview/notificationhubs.json:1555 in e22e3e8. [](commit_id = e22e3e815a5ff351e5ed62bb88e4d62545f38b05, deletion_comment = False)

ms-zhenhua commented:
>         "operationId": "PrivateEndpointConnections_Update",

why "Update", not "Create" or "CreateOrUpdate"?

---
Refers to: specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/preview/2023-01-01-preview/notificationhubs.json:1556 in e22e3e8. [](commit_id = e22e3e815a5ff351e5ed62bb88e4d62545f38b05, deletion_comment = False)

ms-zhenhua commented:
>           "202": {

LRO put should also use 200&201:
[https://armwiki.azurewebsites.net/api_contracts/guidelines/rpaas.html#rpaas001-rpaas-supports-201--provisioning-state--azure-asyncoperation-pattern-only-for-put-long-running-operations](https://armwiki.azurewebsites.net/api_contracts/guidelines/rpaas.html#rpaas001-rpaas-supports-201--provisioning-state--azure-asyncoperation-pattern-only-for-put-long-running-operations)

---
Refers to: specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/preview/2023-01-01-preview/notificationhubs.json:1596 in e22e3e8. [](commit_id = e22e3e815a5ff351e5ed62bb88e4d62545f38b05, deletion_comment = False)

ms-zhenhua commented:
>           "final-state-via": "location"

should use "azure-async-operation":
[https://armwiki.azurewebsites.net/api_contracts/guidelines/rpaas.html#rpaas001-rpaas-supports-201--provisioning-state--azure-asyncoperation-pattern-only-for-put-long-running-operations](https://armwiki.azurewebsites.net/api_contracts/guidelines/rpaas.html#rpaas001-rpaas-supports-201--provisioning-state--azure-asyncoperation-pattern-only-for-put-long-running-operations)

---
Refers to: specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/preview/2023-01-01-preview/notificationhubs.json:1621 in e22e3e8. [](commit_id = e22e3e815a5ff351e5ed62bb88e4d62545f38b05, deletion_comment = False)

ms-zhenhua commented:
>           "200": {

LRO DELETE use 202&204 only:
[https://github.com/Azure/azure-openapi-validator/blob/main/docs/delete-response-codes.md](https://github.com/Azure/azure-openapi-validator/blob/main/docs/delete-response-codes.md)

---
Refers to: specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/preview/2023-01-01-preview/notificationhubs.json:1651 in e22e3e8. [](commit_id = e22e3e815a5ff351e5ed62bb88e4d62545f38b05, deletion_comment = False)

kmiecikt committed:
Notification Hubs: renaming $count to $top

kmiecikt commented:
> > ```
> >     "summary": "Approves or rejects Private Endpoint Connection.\r\nThis is a public API that can be called directly by Notification Hubs users.",
> > ```
> 
> not create?
> 
> Refers to: specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/preview/2023-01-01-preview/notificationhubs.json:1555 in [e22e3e8](https://github.com/Azure/azure-rest-api-specs/commit/e22e3e815a5ff351e5ed62bb88e4d62545f38b05). [](commit_id = [e22e3e8](https://github.com/Azure/azure-rest-api-specs/commit/e22e3e815a5ff351e5ed62bb88e4d62545f38b05), deletion_comment = False)

This is one of the APIs designed by Networking Team and required for Private Link.

The Private Endpoint Connections cannot be created by calling directly our RP. Instead::
1. The user calls the Networking RP method, which verifies the request.
2. The Networking RP calls our RP (using internal API), and we create the Private Endpoint Connection on behalf of the customer.
3. Users can only update (i.e. approve or reject) the Private Endpoint Connections using this PUT API, but they can never create one.

kmiecikt committed:
Notification Hubs: removing 200 response from PrivateEndpointConnection.Delete

kmiecikt committed:
Notification Hubs: fixing async operation annotations

kmiecikt commented:
> > ```
> >       "final-state-via": "location"
> > ```
> 
> should use "azure-async-operation": https://armwiki.azurewebsites.net/api_contracts/guidelines/rpaas.html#rpaas001-rpaas-supports-201--provisioning-state--azure-asyncoperation-pattern-only-for-put-long-running-operations
> 
> Refers to: specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/preview/2023-01-01-preview/notificationhubs.json:1621 in [e22e3e8](https://github.com/Azure/azure-rest-api-specs/commit/e22e3e815a5ff351e5ed62bb88e4d62545f38b05). [](commit_id = [e22e3e8](https://github.com/Azure/azure-rest-api-specs/commit/e22e3e815a5ff351e5ed62bb88e4d62545f38b05), deletion_comment = False)

Fixed

kmiecikt committed:
Merge branch 'main' of https://github.com/Azure/azure-rest-api-specs into kmiecikt-notificationhubs-Microsoft.NotificationHubs-2023-01-01-preview

ms-zhenhua commented:
>               "$ref": "#/definitions/NotificationHubResource"

ProvisioningState property must be defined for LRO PUT:
[https://github.com/Azure/azure-openapi-validator/blob/main/docs/lro-put-provisioning-state-specified.md](https://github.com/Azure/azure-openapi-validator/blob/main/docs/lro-put-provisioning-state-specified.md)

---
Refers to: specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/preview/2023-01-01-preview/notificationhubs.json:165 in e98e9f9. [](commit_id = e98e9f96e97f7f89487a3fe8d17a77b16c3185f9, deletion_comment = False)

ms-zhenhua commented:
>               "$ref": "#/definitions/SharedAccessAuthorizationRuleResource"

ProvisioningState property must be defined in LRO PUT response schema:
[https://github.com/Azure/azure-openapi-validator/blob/main/docs/lro-put-provisioning-state-specified.md](https://github.com/Azure/azure-openapi-validator/blob/main/docs/lro-put-provisioning-state-specified.md)


---
Refers to: specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/preview/2023-01-01-preview/notificationhubs.json:453 in e98e9f9. [](commit_id = e98e9f96e97f7f89487a3fe8d17a77b16c3185f9, deletion_comment = False)

ms-zhenhua commented:
>               "$ref": "#/definitions/SharedAccessAuthorizationRuleResource"

ProvisioningState property must be defined in LRO PUT response schema:
[https://github.com/Azure/azure-openapi-validator/blob/main/docs/lro-put-provisioning-state-specified.md](https://github.com/Azure/azure-openapi-validator/blob/main/docs/lro-put-provisioning-state-specified.md)


---
Refers to: specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/preview/2023-01-01-preview/notificationhubs.json:1205 in e98e9f9. [](commit_id = e98e9f96e97f7f89487a3fe8d17a77b16c3185f9, deletion_comment = False)

ms-zhenhua commented:
>         "subscriptionId": {

It causes the following error and must be deleted:
[https://github.com/Azure/azure-openapi-validator/blob/main/docs/repeated-path-info.md](https://github.com/Azure/azure-openapi-validator/blob/main/docs/repeated-path-info.md)
If it inherits from the old version, please follow this guidance to suppress this error:
[https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/85/Swagger-Suppression-Process](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/85/Swagger-Suppression-Process)

---
Refers to: specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/preview/2023-01-01-preview/notificationhubs.json:2541 in e98e9f9. [](commit_id = e98e9f96e97f7f89487a3fe8d17a77b16c3185f9, deletion_comment = False)

kmiecikt committed:
NotificationHubs: suppressing linter error

kmiecikt committed:
Notification Hubs: fixing async operation annotations

kmiecikt committed:
Notification Hubs: fixing async operation annotations

kmiecikt committed:
Notification Hubs: removing 201 for three operations

kmiecikt commented:
> > ```
> >     "subscriptionId": {
> > ```
> 
> It causes the following error and must be deleted: https://github.com/Azure/azure-openapi-validator/blob/main/docs/repeated-path-info.md If it inherits from the old version, please follow this guidance to suppress this error: https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/85/Swagger-Suppression-Process
> 
> Refers to: specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/preview/2023-01-01-preview/notificationhubs.json:2541 in [e98e9f9](https://github.com/Azure/azure-rest-api-specs/commit/e98e9f96e97f7f89487a3fe8d17a77b16c3185f9). [](commit_id = [e98e9f9](https://github.com/Azure/azure-rest-api-specs/commit/e98e9f96e97f7f89487a3fe8d17a77b16c3185f9), deletion_comment = False)

I added suppression for this one. The `subscriptionId` property was already defined in older API versions.

kmiecikt committed:
Notification Hubs: adding back response 201 for Notification Hubs - Create or Update

kmiecikt committed:
Notification Hubs: cleanup in 201 status codes

kmiecikt commented:
> > ```
> >           "$ref": "#/definitions/SharedAccessAuthorizationRuleResource"
> > ```
> 
> ProvisioningState property must be defined in LRO PUT response schema: https://github.com/Azure/azure-openapi-validator/blob/main/docs/lro-put-provisioning-state-specified.md
> 
> Refers to: specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/preview/2023-01-01-preview/notificationhubs.json:453 in [e98e9f9](https://github.com/Azure/azure-rest-api-specs/commit/e98e9f96e97f7f89487a3fe8d17a77b16c3185f9). [](commit_id = [e98e9f9](https://github.com/Azure/azure-rest-api-specs/commit/e98e9f96e97f7f89487a3fe8d17a77b16c3185f9), deletion_comment = False)

Fixed (also for other comments regarding `ProvisioningState`).
These operations were incorrectly marked as `x-ms-long-running-operation`: although they return 201, they are always synchronous.

zizw123 commented:
Approved the suppression for the repeat path lint error, as it existed in previous stable version.

v-xuto commented:
/azp run

v-xuto commented:
@qiaozha Please help review JS breaking change.
@raych1 Please help review Go breaking change. 
@msyyc Please help review Python breaking change.
@live1206 Please help review `azure-sdk-for-net-track2` error.

live1206 commented:
> Please help review `azure-sdk-for-net-track2` error.

It is failing due to breaking changes.

v-xuto commented:
@qiaozha Please help review JS breaking change.
@raych1 Please help review Go breaking change. 
@msyyc Please help review Python breaking change.

v-xuto commented:
@kmiecikt Please resolve the conflicting files.

kmiecikt committed:
Merge branch 'main' of https://github.com/Azure/azure-rest-api-specs into kmiecikt-notificationhubs-Microsoft.NotificationHubs-2023-01-01-preview

v-xuto commented:
@kmiecikt Please fix CI checks errors (`Model validation`), you can refer to this [comments](https://github.com/Azure/azure-rest-api-specs/pull/24072#issuecomment-1556739784) or  https://aka.ms/ci-fix on how to fix it.

kmiecikt committed:
Notification Hubs: updating ids in examples

kmiecikt committed:
Notification Hubs: updating two examples

kmiecikt commented:
> @kmiecikt Please fix CI checks errors (`Model validation`), you can refer to this [comments](https://github.com/Azure/azure-rest-api-specs/pull/24072#issuecomment-1556739784) or https://aka.ms/ci-fix on how to fix it.

Done. What are the next steps?

v-xuto commented:
@kmiecikt Please resolve the conflicting files.

v-xuto commented:
@chidozieononiwu Please help review the `SwaggerApiView` error in this PR.

kmiecikt committed:
Merge branch 'main' of https://github.com/Azure/azure-rest-api-specs into kmiecikt-notificationhubs-Microsoft.NotificationHubs-2023-01-01-preview

chidozieononiwu commented:
> @chidozieononiwu Please help review the `SwaggerApiView` error in this PR.

I don't see any SwaggerAPIView error. Can you link to what you are referring to.

v-xuto commented:
> > @chidozieononiwu Please help review the `SwaggerApiView` error in this PR.
> 
> I don't see any SwaggerAPIView error. Can you link to what you are referring to.

Re-run pipeline, this issue is no longer present. Thanks for your help.

v-xuto commented:
@rkmanda Please help review the SDK breaking change.

kazrael2119 commented:
this js sdk breaking is caused by this pr

rkmanda reviewed (approved):
:shipit:
